R/dataframe.R

```diff
@@ -182,24 +182,65 @@ as_data_frame.matrix <- function(x, ...) {
   as_data_frame(as.data.frame(x))
 }
 
-#' Convert row names to an explicit variable.
+#' Conversion between rownames and a column in data frame
+#'
+#' \code{add_rownames or rownames_to_column} convert row names to an explicit variable.
 #'
 #' @param df Input data frame with rownames.
 #' @param var Name of variable to use
 #' @export
+#' @rdname rownames
 #' @examples
 #' tbl_df(mtcars)
 #'
-#' add_rownames(mtcars)
+#' (mtcars_tbl <- add_rownames(mtcars) )
+#'
+#' rownames_to_column(mtcars)
 add_rownames <- function(df, var = "rowname") {
   stopifnot(is.data.frame(df))
 
+  if (var %in% colnames(df) ) {
+    stop(paste("There is a column named", var, "already!"))
+  }
+
   rn <- as_data_frame(setNames(list(rownames(df)), var))
   rownames(df) <- NULL
 
   as_data_frame(cbind(rn, df))
+
 }
 
+#' @rdname rownames
+#' @export
+rownames_to_column <- add_rownames
```

Comment:
`rownames_to_column` is a copy of `add_rownames`. So you mean keep the input of `rownames_to_column` as it is (I did not change that section)? For example, input `data.frame` output `data.frame`? I will remove `add_rownames`.